Musicca/app/src/main/java/com/example/musicca/fragments/MultiFilterDialogFragment.java

```diff
@@ -0,0 +1,106 @@
+package com.example.musicca.fragments;
+
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.AdapterView;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import com.example.musicca.R;
+
+public class MultiFilterDialogFragment extends DialogFragment implements AdapterView.OnItemSelectedListener {
+
+    private static final String TITLE_ARG = "title";
+
+    private String YEAR_SELECTED;
+    private TextView yearReleased;
+    private Spinner yearSpinner;
+    private TextView tvSongTitle;
+    private EditText etSongTitle;
+    private TextView tvSongArtist;
+    private EditText etSongArtist;
+    private Button btnApplyFilter;
+
+    public MultiFilterDialogFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+        YEAR_SELECTED = adapterView.getSelectedItem().toString();
```

Comment:
Consider making this filter even more powerful by supporting range in year, like 2000 <= year <= 2010. Doesn't have to happen in this PR.